### PR TITLE
allow arguments with short flag only long optional

### PR DESCRIPTION
**Pull request by syrshax**
as detailed in the issue #123 now tools can no longer have a limitation of forcing long flags

### CODE REVIEW

File: argparse_test.go [hunk 1]
```diff
@@ -92,7 +92,6 @@ func TestFlagAddArgumentFail(t *testing.T) {
 	}
 	tt := []testCase{
 		testCase{testName: "Long short name", shortArg: "ff", longArg: "flag2", failureMessage: "unable to add Flag: short name must not exceed 1 character"},
-		testCase{testName: "Long name not provided", shortArg: "f", longArg: "", failureMessage: "unable to add Flag: long name should be provided"},
```
Comment (jlombera) [inline argparse_test.go:95]: Perhaps also add case that test that an error is produced if both short and long names are empty (??).

File: argparse_test.go [hunk 2]
```diff
@@ -3231,3 +3219,28 @@ func TestPositionalDefaults(t *testing.T) {
 		t.Errorf(`*pos4 expected "pos4", but got "%s"`, *pos4)
 	}
 }
+
+func TestOptionalLongArgSuccess(t *testing.T) {
```
Comment (jlombera) [inline argparse_test.go]: I was thinking in a test that tries all the combinations:

- non-empty short name, non-empty long name: success
- empty short name, non-empty long name: success
- non-empty short name, empty long name: success
- empty short name, empty long name: error

Comment (syrshax) [reply argparse_test.go]: Yes indeed I would! 

This is one of my first contributions since I'm kind of new to open source, I'll make a proper table case with those combinations,  thank you very much for the feedback!!

File: argparse_test.go (hunk 1 above)
Comment (syrshax) [reply argparse_test.go:95]: I added for all the type cases the error of both args being empty

File: argparse_test.go [hunk 3]
```diff
@@ -92,7 +92,7 @@ func TestFlagAddArgumentFail(t *testing.T) {
 	}
 	tt := []testCase{
 		testCase{testName: "Long short name", shortArg: "ff", longArg: "flag2", failureMessage: "unable to add Flag: short name must not exceed 1 character"},
-		testCase{testName: "Long name not provided", shortArg: "f", longArg: "", failureMessage: "unable to add Flag: long name should be provided"},
+		testCase{testName: "Both names empty", shortArg: "", longArg: "", failureMessage: "unable to add Flag: either long name or short name should be provided"},
```
Comment (jlombera) [inline argparse_test.go]: Just to make it consistent with similar changes to all other tests, move this case to the end of the list.

File: argparse_test.go [hunk 4]
```diff
@@ -3231,3 +3231,33 @@ func TestPositionalDefaults(t *testing.T) {
 		t.Errorf(`*pos4 expected "pos4", but got "%s"`, *pos4)
 	}
 }
+
+func TestOptionalLongArg(t *testing.T) {
```
Comment (jlombera) [inline argparse_test.go]: Minor nit, but I think this test should be somewhere around the beginning of the file.  It precludes other tests in the sense that once it's being tested that long/short names are optional (but not both), you can write tests with empty args when it's not important (e.g. `TestShortFlagEqualChar()`, `TestFlagMultiShorthand1()`, etc).  Not that should changes those tests in this PR, but it gives a nice mental model when you read this test file.

File: argparse_test.go (hunk 3 above)
Comment (syrshax) [reply argparse_test.go]: Absolutely, change made

File: argparse_test.go (hunk 4 above)
Comment (syrshax) [reply argparse_test.go]: moved the test right after ```TestInternalFunctionCheck()``` this way it establishes the fundamental validation rule before the type-specific tests that assume it

File: command.go [hunk 5]
```diff
@@ -26,8 +26,8 @@ func (o *Command) help(sname, lname string) {
 
 func (o *Command) addArg(a *arg) error {
 	// long name should be provided
```
Comment (jlombera) [inline command.go]: Update this comment too.

**Review by jlombera — COMMENTED** [review]
I don't have any say on this (not owner/maintainer/contributor) but LGTM.

File: argparse.go
Comment (jlombera) [inline argparse.go]: This needs update too.  Currently the help message when the long name is empty is printed this this:
```
  -s  --       <Help string>
```

Following change fixed it for me:
```go
if argument.lname != "" {
    arg = arg + "--" + argument.lname
}
```
Which results in following message:
```
  -s           <Help string>
```

Comment (syrshax) [reply argparse.go]: Thank you again for the feedback!
